### PR TITLE
Fit BarqCurve control points to curve #5

### PR DESCRIPTION
**Pull request by Kitsunp**
# Issue #5 Resolution: Fit BarqCurve control points to curve

## Changes made to optspacecurve.py:

1.  Added `_fit_curve_to_control_points()` method using least squares + Bernstein matrix
    Uses JAX's `jnp.linalg.pinv()` for optimal performance with existing `beziertools` functions

2.  Enhanced `BarqCurve.initialize_parameters()` with 3 new optional parameters:
    *   `fit_target_curve`: function(t) -> `[x,y,z]` for curve fitting
    *   `fit_target_points`: array of target points for fitting
    *   `fit_n_samples`: number of samples for parameterization

3.  Maintains 100% backward compatibility - existing code works unchanged
    `barq_curve.initialize_parameters(seed=42)`  # Still works as before

4.  Enables informed initialization instead of random points, improving optimization performance
    `barq_curve.initialize_parameters(fit_target_curve=lambda t: [cos(t), sin(t), 0])`

### CODE REVIEW

**Comment by Kitsunp** [comment]
- Add target curve/points fitting in initialize_parameters()
- Refactor to use existing prepare_optimization_loss + optimize infrastructure
- Improve parameter validation and error handling with proper error raising
- Enhance documentation with examples and use cases
- Maintain full backward compatibility
- Add comprehensive tests for new functionality

Incorporates @gderossi feedback:
- Use existing optimization infrastructure instead of manual gradient descent
- Proper pgf/prs parameter propagation
- Strict validation with meaningful error messages
- Simplified and focused test suite

**Review by gderossi — CHANGES_REQUESTED** [review]
Hi @Kitsunp, thanks for those changes! I have two more tiny tweaks for your tests (most of which is copy/pasting or deleting), and then this will be good to merge!

**Comment by Kitsunp** [comment]
@gderossi, the mentioned corrections have already been made.

**Comment by gderossi** [comment]
Thanks @Kitsunp, we've looked this over and it seems good to go!
